### PR TITLE
[LLVM] Upgrade to LLVM 13.0.0

### PR DESCRIPTION
**Pull request by vchuravy**
https://github.com/llvm/llvm-project/compare/llvmorg-13.0.0...JuliaLang:julia-13.0.0-3

### CODE REVIEW

**Comment by vtjnash** [comment]
Now just waiting on https://reviews.llvm.org/D114389, and getting LLVM (re-)packaged correctly in the build for Win32/Win64.

File: base/binaryplatforms.jl [hunk 1]
```diff
@@ -894,7 +894,7 @@ function detect_cxxstring_abi()
     end
 
     function open_libllvm(f::Function)
-        for lib_name in ("libLLVM", "LLVM", "libLLVMSupport")
+        for lib_name in ("libLLVM", "LLVM", "libLLVMSupport", "libLLVM-13jl")
```
Comment (staticfloat) [inline base/binaryplatforms.jl:897]: Looking through the rest, if we know it's generally going to be called `libLLVM-13jl`, I think we should try to dlopen that one first, and only fall back to `libLLVM` afterward.

**Review by staticfloat — APPROVED** [review]

**Comment by vchuravy** [comment]
@nanosoldier `runtests(ALL, vs = ":master")`

**Comment by vchuravy** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by nanosoldier** [comment]
[Your package evaluation job](https://github.com/JuliaLang/julia/pull/42602#issuecomment-988980833) has completed - possible new issues were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/pkgeval/by_hash/d24d243_vs_8197c41/report.md).

**Comment by vchuravy** [comment]
@nanosoldier `runbenchmarks(ALL, vs=":master")`

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/42602#issuecomment-1001744567):
```
NanosoldierError: error when preparing/pushing to report repo: failed process: Process(setenv(`git push`; dir="/data/nanosoldier/workdir/NanosoldierReports"), ProcessExited(1)) [1]

```
Unfortunately, the logs could not be uploaded.

**Comment by vtjnash** [comment]
https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/15db64c_vs_c790938/report.md Looks okay. Sorry that this new AWS machine seems to be unreliable. Hopefully we can move back to dedicated hardware soon.

@nanosoldier `runbenchmarks("collections" || "shootout" || "dates" || "string", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/42602#issuecomment-1001867036) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/15db64c_vs_c790938/report.md).

**Comment by vchuravy** [comment]
Oh yeah that's extremely noisy :/

**Comment by vchuravy** [comment]
Non-scientific build benchmark:

### Before:
```
Sysimage built. Summary:
Total ───────  52.908324 seconds 
Base: ───────  21.431235 seconds 40.5064%
Stdlibs: ────  31.475813 seconds 59.4912%
    JULIA /home/vchuravy/builds/julia/usr/lib/julia/sys-o.a
Generating REPL precompile statements... 36/36
Executing precompile statements... 1309/1340
Precompilation complete. Summary:
Total ─────── 109.812820 seconds
Generation ──  82.326156 seconds 74.9695%
Execution ───  27.486664 seconds 25.0305%
    LINK /home/vchuravy/builds/julia/usr/lib/julia/sys.so
```

### After:
```
Sysimage built. Summary:
Total ───────  52.104447 seconds 
Base: ───────  20.850621 seconds 40.017%
Stdlibs: ────  31.252741 seconds 59.9809%
    JULIA /home/vchuravy/builds/julia/usr/lib/julia/sys-o.a
Generating REPL precompile statements... 36/36
Executing precompile statements... 1309/1340
Precompilation complete. Summary:
Total ─────── 102.547306 seconds
Generation ──  75.177300 seconds 73.3099%
Execution ───  27.370006 seconds 26.6901%
```

**Pull request merged**